logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultIteratorAssertions.kt

```diff
@@ -14,4 +14,17 @@ class DefaultIteratorAssertions : IteratorAssertions {
 
     override fun <E, T : Iterator<E>> hasNotNext(container: AssertionContainer<T>): Assertion =
         container.createDescriptiveAssertion(HAS_NOT, NEXT_ELEMENT) { !it.hasNext() }
+
+    override fun <E, T : Iterator<E>> containsNoDuplicates(container: AssertionContainer<T>): Assertion {
+        val finalList = mutableListOf<E>()
+
+        return container.createDescriptiveAssertion(CONTAINS_DUPLICATES, DUPLICATE_FOUND) {
+            if(finalList.isEmpty()){
```

Comment:
why this check? it is always empty no? maybe you wanted to write `it.iterator.hasNext()`? 
This should not be necessary, addAll covers that

Reply:
For some reason the body of the 'test' which is in {}, is executed multiple times because there is methods holds() which check the boolean value. Because it is iterable and not collection we are allow to go over it only once so I keep the collection as a copy on a call of this method to avoid re-iteration. Now is make sense?

Reply:
Makes sense